tests/bcl-test/BCLTests/templates/iOSApp/ViewController.cs

```diff
@@ -75,17 +75,36 @@ public async override void ViewDidLoad ()
 			var logger = (writer == null || options.EnableXml) ? new LogWriter () : new LogWriter (writer);
 			logger.MinimumLogLevel = MinimumLogLevel.Info;
 			var testAssemblies = GetTestAssemblies ();
-			Xamarin.iOS.UnitTests.TestRunner runner;
-			if (RegisterType.IsXUnit)
-				runner = new XUnitTestRunner (logger);
-			else {
-				runner = new NUnitTestRunner (logger) {
-					// add known ignored categories.
-					Filter = new NotFilter (new CategoryExpression ("MobileNotWorking,NotOnMac,NotWorking,ValueAdd,CAS,InetAccess,NotWorkingLinqInterpreter").Filter)
+			var runner = RegisterType.IsXUnit ? (Xamarin.iOS.UnitTests.TestRunner) new XUnitTestRunner (logger) : new NUnitTestRunner (logger);
+			var categories = RegisterType.IsXUnit ?
+				new List<string> { 
+					"failing",
+					"nonmonotests",
+					"outerloop",
+					"nonosxtests"
+				} :
+				new List<string> {
+					"MobileNotWorking",
+					"NotOnMac",
+					"NotWorking",
+					"ValueAdd",
+					"CAS",
+					"InetAccess",
+					"NotWorkingLinqInterpreter",
```

Comment:
DRY
can we move this (or eve more) code to some shared code ?
it will get out-of-sync (it's a matter of time)

Reply:
And you read my mind and @akoeplinger 

We talked about this being moved to the binary artifacts from mono. It is listed in https://github.com/mono/mono/issues/11397 as one of the needed changes for us :)